### ISSUE TITLE
Notice: Array to string conversion when using array in @QueryParam requirements option

### ISSUE THREAD

**Issue by kwn** [nelmio/NelmioApiDocBundle#564]
Hi guys,

I found a bug. While I'm trying to visit documentation pages I get an exception: 
```
Notice: Array to string conversion in /var/www/vhosts/localhost.crkl/httpdocs/vendor/nelmio/api-doc-bundle/Nelmio/ApiDocBundle/Extractor/Handler/FosRestHandler.php line 86

500 Internal Server Error - ContextErrorException 
```

My annotations:

```php
/**
 * @Rest\RouteResource("Venue")
 */
class VenueController extends FOSRestController
{
    /**
     * @param ParamFetcher $paramFetcher
     * @return array
     *
     * @Rest\View()
     * @Rest\QueryParam(name="lat", strict=true)
     * @Rest\QueryParam(name="lng", strict=true)
     * @Rest\QueryParam(name="range", strict=true, requirements={
     *     @Assert\GreaterThan("0"),
     *     @Assert\Regex("/\d+/")
     * }, default="10")
     * @Rest\QueryParam(name="product", strict=true, requirements={
     *     @Assert\Regex("/\d+/")
     * }, nullable=true)
     * @ApiDoc()
     */
    public function cgetAction(ParamFetcher $paramFetcher)
    {
        // [...]
    }
```

So, the reason of this error is an array in ```requirements``` option. After removing ```range``` QueryParam and change ```product``` QueryParam to:

```
@Rest\QueryParam(name="product", strict=true, requirements=@Assert\Regex("/\d+/"), nullable=true)
```

everyting works fine.

**Comment by solazs** [comment]
Same here with

```
/**
     * @Annotations\QueryParam(name="offset", requirements="\d+", nullable=true, description="Offset from which to start listing news.")
     * @Annotations\QueryParam(name="limit", requirements="\d+", default=null, description="How many news to return.")
     * @Annotations\QueryParam(name="tags", array=true, default={}, description="Tags to filter the query")
     * @Annotations\QueryParam(name="services", array=true, default={}, description="Service IDs to filter the query")
     * @Annotations\QueryParam(name="organizations", array=true, default={}, description="Organization IDs to filter the query")
     * @ApiDoc(
     *   section = "News",
     *   resource = true,
     *   description = "get news for the current user",
     *   statusCodes = {
     *     200 = "Returned when successful",
     *     401 = "Returned when token is expired or invalid",
     *     403 = "Returned when not permitted to query",
     *     404 = "Returned when resource is not found"
     *   },
     *   requirements ={
     *      {"name"="_format", "requirement"="xml|json", "description"="response format"}
     *   }
     * )
     */
```

**Comment by willdurand** [comment]
@advancingu could you open a PR with your fix please?

**Comment by advancingu** [comment]
I did but you closed it due to lack of tests (understandable). I simply don't have the time for it.
https://github.com/nelmio/NelmioApiDocBundle/pull/596

**Comment by willdurand** [comment]
Should be fixed in [2.11.2](https://github.com/nelmio/NelmioApiDocBundle/releases/tag/2.11.2).

**Comment by zhukovra** [comment]
@willdurand, no, it shouldn't. You can see it in my PR #796, where new test is added, with plain array in QueryParam requirements:
```
@QueryParam(name="param1", requirements={@Assert\NotNull(), @Assert\NotBlank()}
```
